src/main.cc

```diff
@@ -253,6 +253,7 @@ int main(int argc, char *argv[])
         exitCode = app->exec();
     }
 
+    app->_shutdown();
```

Comment:
Nothing in toolbox or main window lives on another thread so that should be ok. I'm going to leave this as is. It still may not be exactly correct. But it's better than before and calling sendPostedEvents may lead to other strange things we've never seen on shutdown.